### PR TITLE
Refactor create shoot test to make it reuseable

### PR DESCRIPTION
**Pull request by timebertt**
**What this PR does / why we need it**:
This PR refactors the `CreateShoot` integration test to make it reusable in other tests.
Other test can now leverage the new `ShootCreationFramework` to easily create a Shoot just like the `CreateShoot` test does.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
cc @schrodit 

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Review by dguendisch — DISMISSED** [review]

File: test/framework/shoot_utils.go [hunk 1]
```diff
@@ -146,106 +146,141 @@ func (f *ShootFramework) DumpState(ctx context.Context) {
...
-	if cloudProfile != nil && len(*cloudProfile) > 0 {
-		shoot.Spec.CloudProfileName = *cloudProfile
+	if StringSet(projectNamespace) {
+		shoot.Namespace = projectNamespace
+	}
+
+	if shoot.Annotations == nil {
+		shoot.Annotations = map[string]string{}
+	}
+	shoot.Annotations[v1beta1constants.AnnotationShootIgnoreAlerts] = "true"
```
Comment (schrodit) [inline test/framework/shoot_utils.go]:
```suggestion
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	metav1.SetMetaDataAnnotation(&shoot.ObjectMeta, v1beta1constants.AnnotationShootIgnoreAlerts, "true")
```

**Review by schrodit — CHANGES_REQUESTED** [review]
Rest is lgtm

File: test/framework/shoot_utils.go (hunk 1 above)
Comment (timebertt) [reply test/framework/shoot_utils.go]: Done, PTAL :)

**Review by schrodit — APPROVED** [review]

**Comment by timebertt** [comment]
Hmm, I can't merge this PR as @gardener/test-infra-maintainers don't have write access to g/g anymore, so their reviews don't count towards mergeability...

**Comment by rfranzke** [comment]
They never had 😉What about now?

**Comment by timebertt** [comment]
Ah, I thought they had, because that's the reason github was telling me, why the checkmark wasn't green 😄

**Pull request merged**